Framework/PythonInterface/plugins/algorithms/SaveP2D.py

```diff
@@ -0,0 +1,202 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import numpy as np
+import math
+from scipy.special import lambertw
+from mantid import mtd
+from mantid.api import (AlgorithmFactory, PythonAlgorithm, FileAction, FileProperty, WorkspaceProperty, Progress)
+from mantid.kernel import Direction
+
+
+class SaveP2D(PythonAlgorithm):
+    def category(self):
+        return 'Diffraction//Data Handling'
+
+    def summary(self):
+        """
+        summary of the algorithm
+        :return:
+        """
+        return "The algorithm used to create a multidimensional '.p2d' file from a 2D workspace."
+
+    def name(self):
+        return "SaveP2D"
+
+    def seeAlso(self):
+        return ["PowderReduceP2D", "Bin2DPowderDiffraction"]
+
+    def PyInit(self):
+        # Input file
+        self.declareProperty(WorkspaceProperty('Workspace',
+                                               '',
+                                               direction=Direction.Input),
+                             doc='Workspace that should be used.')
+        # Output File
+        self.declareProperty(FileProperty('OutputFile',
+                                          '',
+                                          action=FileAction.Save,
+                                          direction=Direction.Input),
+                             doc='Output File for ".p2d" Data.')
+        # Manipulating Data ranges
+        self.declareProperty(
+            'RemoveNaN',
+            True,
+            direction=Direction.Input,
+            doc='Remove DataPoints with NaN as intensity value')
+        self.declareProperty(
+            'RemoveNegatives',
+            True,
+            direction=Direction.Input,
+            doc='Remove data points with negative intensity values')
+        self.declareProperty(
+            'CutData',
+            False,
+            direction=Direction.Input,
+            doc=
+            'Use the following inputs to limit data in Theta, lambda, d and dp'
+        )
+        self.declareProperty('tthMin',
```

Comment:
Okay so the last test we are failing on is coding standards, it wants these properties to start with a capital letter and each new word the in the variable name should also start with a captial letter. So this should be
`TthMin` or `TTHMin`
You'll have to then change the rest of the places that use this variable in the code